src/cpp/r/RSexp.cpp

```diff
@@ -172,24 +187,61 @@ SEXP findVar(const std::string& name, const std::string& ns)
    
    return findVar(name, env);
 }
-  
+
 SEXP findFunction(const std::string& name, const std::string& ns) 
 {
+   r::sexp::Protect protect;
    if (name.empty())
       return R_UnboundValue;
    
    SEXP env = ns.empty() ? R_GlobalEnv : findNamespace(ns);
+   if (env == R_UnboundValue) return R_UnboundValue;
    
-   SEXP functionSEXP;
-   Error error = executeSafely<SEXP>(
-      boost::bind(Rf_findFun, Rf_install(name.c_str()), env), 
-      &functionSEXP
-   );
+   // We might want to use `Rf_findFun`, but it calls `Rf_error`
+   // on failure, which involves printing the error message out
+   // to the console. To avoid this,
+   // we instead attempt to find the function by manually
+   // walking through the environment (and its enclosing environments)
+   SEXP nameSEXP = Rf_install(name.c_str());
    
-   if (error)
-      return R_UnboundValue;
-   else
-      return functionSEXP ;
+   // Search through frames until we find the global environment.
+   while (env != R_EmptyEnv)
+   {
+      // If we're searching the global environment, then
+      // try using 'Rf_findVar', as this will attempt a search
+      // of R's own internal global cache.
+      if (env == R_GlobalEnv)
+      {
+         SEXP resultSEXP = Rf_findVar(nameSEXP, R_GlobalEnv);
+         if (Rf_isFunction(resultSEXP))
+            return resultSEXP;
+         else if (TYPEOF(resultSEXP) == PROMSXP)
+         {
+            protect.add(resultSEXP = Rf_eval(resultSEXP, env));
```

Comment:
If we are worried about side effects or errors being thrown then the way around this is r::exec::executeSafely<SEXP> which will put a box around any of these errors.